File: components/credentials/src/main/java/org/cloudfoundry/credhub/views/PasswordView.java
package org.cloudfoundry.credhub.views;

import org.cloudfoundry.credhub.credential.StringCredentialValue;
import org.cloudfoundry.credhub.domain.PasswordCredentialVersion;

@SuppressWarnings("unused")
public class PasswordView extends CredentialView {

  public PasswordView() {
    super();
  }

  public PasswordView(final PasswordCredentialVersion passwordCredential) {
    super(
      passwordCredential.getVersionCreatedAt(),
      passwordCredential.getUuid(),
      passwordCredential.getName(),
      passwordCredential.getCredentialType(),
      new StringCredentialValue(passwordCredential.getPassword())
    );
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/views/PasswordView.kt
package org.cloudfoundry.credhub.views

import org.cloudfoundry.credhub.credential.StringCredentialValue
import org.cloudfoundry.credhub.domain.PasswordCredentialVersion

class PasswordView : CredentialView {
    constructor() : super() {}
    constructor(passwordCredential: PasswordCredentialVersion) : super(
        passwordCredential.versionCreatedAt,
        passwordCredential.uuid,
        passwordCredential.name,
        passwordCredential.getCredentialType(),
        passwordCredential.metadata,
        StringCredentialValue(passwordCredential.password)
    ) {
    }
}
